src/components/Teaser/Teaser.tsx

```diff
@@ -1,104 +1,182 @@
 import * as React from 'react';
 import { bem } from '../../utils';
 import { Text } from '../Text';
-import { Tooltip } from '../Tooltip';
+import { Checkbox } from '../Checkbox';
+import { Link } from '../Link';
+import { TagProps } from '../Tag';
 import styles from './Teaser.scss';
 
-type Status = {
-    label: string;
-    tooltip: string;
+type Info = {
+    text: string;
+    href?: string;
+    icon?: React.ReactElement;
 };
 
 export interface Props extends Omit<React.HTMLAttributes<HTMLDivElement>, 'title'> {
     /** The title of entity. Note: it is rendered within a <span> */
-    title: React.ReactNode;
+    title: string;
     /** The subtitle of the entity. Note: it is rendered within a <span> */
-    subTitle?: React.ReactNode;
-    /** The location of the entity. Note: it is rendered within a <span> */
-    location?: React.ReactNode;
-    /** Details related to the entity. Note: it is rendered within a <span> */
-    details?: React.ReactNode;
-    /** Statuses of entity containing status label and tooltip message */
-    statuses?: Status[];
+    subtitle?: string;
     /** Style the component as disabled */
     disabled?: boolean;
+    /** Define if checkbox is rendered or not */
+    hasCheckbox?: boolean;
+    /** Checkbox id */
+    id?: string;
+    /** Checkbox status */
+    isSelected?: boolean;
+    /** When checkbox status changes */
+    onChange?: React.ChangeEventHandler<HTMLInputElement> | undefined;
+    /** First line of information */
+    primaryInfo?: Info;
+    /** Second line of information */
+    secondaryInfo?: Info;
+    /** Third line of information */
+    tercearyInfo?: Info;
+    /** Line containing the source of the information */
+    sourceInfo?: Info;
+    /** Render little blue bubble */
+    isVisited?: boolean;
+    /** Render timestamp information */
+    timestamp?: string;
+    /** Render tags */
+    tags?: React.ReactElement<TagProps>[];
+    /** Matching indicator percentage */
+    matchingIndicatorPercentage?: number;
 }
 
 const { block, elem } = bem('Teaser', styles);
 
+const InfoComponent = ({ text, href, icon }: Info) => {
```

Comment:
I see 2 patters in the component, one reuses `textComponent` perhaps with additional classes using React.cloneElement, two writing the text component from scratch. I'm good with either by perhaps not with both, if you see what I mean. 

Also, maybe it makes sense to move it to a separate file, so that it is easier to test?